operator/v1/types.go

```diff
@@ -53,6 +53,19 @@ type ClusterManagerSpec struct {
 	// +optional
 	// +kubebuilder:default={mode: Default}
 	DeployOption ClusterManagerDeployOption `json:"deployOption,omitempty"`
+
+	// RegistrationConfiguration contains the configuration of registration
+	// +optional
+	RegistrationConfiguration RegistrationConfiguration `json:"registrationConfiguration,omitempty"`
+}
+
+type RegistrationConfiguration struct {
+	// FeatureGates represents the list of feature gates for registration
+	// If it is set empty, default feature gates will be used.
+	// If it is set not empty, registration controller will use all items of it as args.
+	// And it should be set like ["featuregate1=true", "featuregate2=false"]
+	// +optional
+	FeatureGates []string `json:"featureGates,omitempty"`
```

Comment:
fully structurelize the expression `a=true` into a string-bool map, in case of injection